google/gax/retry.py

```diff
@@ -142,6 +142,16 @@ def inner(*args):
                     timeout = min(
                         timeout * timeout_mult, max_timeout, deadline - now)
 
-        raise exc
+                # If we have passed our deadline, raise the exception.
+                if deadline is not None and now >= deadline:
+                    exc = errors.RetryError(
+                        'Retry timeout exceeded; exception: %s' % exception,
+                        cause=exception,
+                    )
+                    six.reraise(errors.RetryError, exc, sys.exc_info()[2])
```

Comment:
@jonparrott Heading to a meeting, so this will be delayed. I have not forgotten it.